### ISSUE TITLE
ARM64 support

### ISSUE THREAD

**Issue by JoelLinn** [zyedidia/perforator#7]
Unfortunately this does not support ARM64 (or any non x86-64 architecture)

**Comment by zyedidia** [comment]
Yes, there are a few x86-64-specific things in the codebase. In particular, the interrupt instruction that is used to return control to perforator when a traced function is entered is architecture-specific (defined [here](https://github.com/zyedidia/perforator/blob/58a20773241493a754e478eab972e54a5ae08bc1/utrace/proc.go#L13)), and the function for determining the return address of a stack frame (defined [here](https://github.com/zyedidia/perforator/blob/58a20773241493a754e478eab972e54a5ae08bc1/utrace/region.go#L44). The code also uses an x86-64 ptrace regs struct. If these parts are updated, I think perforator could work on ARM as well, but I have no plans to implement this myself.
